mlflow/utils/__init__.py

```diff
@@ -272,7 +272,7 @@ def get_results_from_paginated_fn(paginated_fn, max_results_per_page, max_result
     return all_results
 
 
-def _insecure_md5(string=b""):
+def _md5_hash(string=b""):
```

Comment:
```suggestion
def _hash(name, string=b""):
```

We might want to add `name` for #10106.

Reply:
What about `_insecure_hash`? I see see value in keeping explicit alert in its name, and if we no longer say "md5" anyway then the original concern ("is it insecure version of md5?") won't be a big matter.

Reply:
I added a `insecure_hash` module instead :)